src/Databases/TablesLoader.cpp

```diff
@@ -76,17 +72,66 @@ void TablesLoader::loadTables()
     /// Remove tables that do not exist
     removeUnresolvableDependencies();
 
-    loadTablesInTopologicalOrder(pool);
-}
+    /// Compatibility setting which should be enabled by default on attach
+    /// Otherwise server will be unable to start for some old-format of IPv6/IPv4 types of columns
+    ContextMutablePtr load_context = Context::createCopy(global_context);
+    load_context->setSetting("cast_ipv4_ipv6_default_on_conversion_error", 1);
+
+    for (const auto & table_id : all_loading_dependencies.getTablesSortedByDependency())
+    {
+        /// Gather tasks to load before this table
+        LoadTaskPtrs load_dependency_tasks;
+        for (const StorageID & dependency_id : all_loading_dependencies.getDependencies(table_id))
+            load_dependency_tasks.push_back(load_table[dependency_id.getFullTableName()]);
+
+        // Make load table task
+        auto table_name = table_id.getQualifiedName();
+        const auto & path_and_query = metadata.parsed_tables[table_name];
+        auto task = databases[table_name.database]->loadTableFromMetadataAsync(
+            async_loader,
+            getGoalsOr(load_dependency_tasks, load_after),
+            load_context,
+            path_and_query.path,
+            table_name,
+            path_and_query.ast,
+            strictness_mode);
+        load_table[table_id.getFullTableName()] = task;
+        result.push_back(task);
+    }
 
+    return result;
+}
 
-void TablesLoader::startupTables()
+LoadTaskPtrs TablesLoader::startupTablesAsync(LoadJobSet startup_after)
 {
-    /// Startup tables after all tables are loaded. Background tasks (merges, mutations, etc) may slow down data parts loading.
-    for (auto & database : databases)
-        database.second->startupTables(pool, strictness_mode);
-}
+    LoadTaskPtrs result;
+    std::unordered_map<String, LoadTaskPtrs> startup_database; /// database name -> all its tables startup tasks
+
+    for (const auto & table_id : all_loading_dependencies.getTablesSortedByDependency())
```

Comment:
We can startup tables in any order, `getTablesSortedByDependency` is not necessary